sandman/source/ncurses_context.cpp

```diff
@@ -0,0 +1,146 @@
+#include "ncurses_context.h"
+
+namespace Sandman::NCurses
+{
+
+	// This window is where messages from the logger are written to.
+	static WINDOW* s_LoggingWindow = nullptr;
+
+	WINDOW* GetLoggingWindow()
+	{
+		return s_LoggingWindow;
+	}
+
+	// This window is where user input is echoed to.
+	static WINDOW* s_InputWindow = nullptr;
+
+	WINDOW* GetInputWindow()
+	{
+		return s_InputWindow;
+	}
+
+	/*
+		Configure a window with "sensible" defaults.
+	*/
+	static void DefaultConfigureWindow(WINDOW* const p_Window)
+	{
+		/* output options: `man 'outopts(3NCURSES)'` */
+		{
+			// Don't make the next call to `wrefresh` clear and redraw the screen completely.
+			clearok(p_Window, FALSE);
+
+			// Terminal's insert line & delete line features are okay to use.
+			idlok(p_Window, TRUE);
+
+			// Terminal's insert character & delete character features are okay to use.
+			idcok(p_Window, TRUE);
+
+			// Every change to the window will not cause a refresh to the physical screen.
+			immedok(p_Window, FALSE);
+
+			// Leave the hardware cursor wherever the update happens to leave it, saving time.
+			leaveok(p_Window, FALSE);
+
+			// Don't scroll when cursor is moved off the edge of the window.
+			scrollok(p_Window, FALSE);
+		}
+
+		/* input options: `man 'inopts(3NCURSES)'` */
+		{
+			// Don't flush everything on interrupts.
+			intrflush(p_Window, FALSE);
+
+			// Enable functon keys (F1, F2, ...), arrow keys, etc.
+			keypad(p_Window, TRUE);
+
+			// Make `getch` non-blocking.
+			nodelay(p_Window, TRUE);
+
+			// Don't set a timer to wait for the next character when interpreting an escape sequence.
+			notimeout(p_Window, TRUE);
+		}
+	}
+
+} // namespace NCurses
+
+void NCurses::Initialize()
+{
+	/*
+		Initialize NCurses. This function exits the program on error!
+		This initializes the standard screen `stdscr` window.
+	*/
+	initscr();
+
+	/* input options: `man 'inopts(3NCURSES)'` */
+	// These options in particular do not take a window pointer.
+	{
+		/*
+			Don't have to wait for newlines in order to get characters.
+			Control characters are treated normally.
+			Control characters on Linux include Ctrl+Z (suspend), Ctrl+C (quit), etc.
+
+			Note: it is not recomended to call `raw` or `noraw` after setting `cbreak` or `nocbreak`
+			according to "NOTES" section of `man 'inopts(3NCURSES)'`.
+		*/
+		cbreak();
+
+		/*
+			Do not echo the characters typed by the user into the terminal by default.
+			Echo of the user input should be implemented manually for this project.
+		*/
+		noecho();
+
+		// No newlines; translate newline characters to carriage return characters.
+		nonl();
+	}
+
+	// Configure standard screen window.
+	DefaultConfigureWindow(stdscr);
+
+	// The input window will have a height of 3.
+	static constexpr int s_InputWindowRowCount{ 3 };
+
+	/* configure logging window */
```

Comment:
Prefer line comments especially for single line comments.

Reply:
Fixed. Now there isn't a single `/*` in `source`.